src/main/java/wooteco/subway/domain/Sections.java

```diff
@@ -0,0 +1,215 @@
+        }
+        return newDistance;
+    }
+
+    private Section insertBeforeDownStation(Section newSection, Section section) {
+        int index = values.indexOf(section);
+        Section sectionToModify = new Section(section.getId(), section.getUpStation(), newSection.getUpStation(),
+            getNewDistance(section, newSection));
+        values.set(index, sectionToModify);
+        values.add(index + 1, newSection);
+        return sectionToModify;
+    }
+
+    private void validateAddable(Section section) {
+        List<Section> foundSections = findSectionsOverlapped(section);
+
+        if (isInvalid(section, foundSections)) {
+            throw new IllegalArgumentException("이미 포함된 두 역을 가진 Section 을 추가할 수 없습니다.");
+        }
+    }
+
+    private List<Section> findSectionsOverlapped(Section newSection) {
+        return values.stream()
+            .filter(section ->
+                section.contains(newSection.getDownStation()) || section.contains(newSection.getUpStation()))
+            .collect(Collectors.toList());
+    }
+
+    private boolean isInvalid(Section section, List<Section> foundSections) {
+        return hasSameStations(section) ||
+            isInvalidCountInMiddle(foundSections) || isSameWithDestinations(foundSections);
+    }
+
+    private boolean hasSameStations(Section newSection) {
+        return values.stream().anyMatch(section -> section.hasSameStations(newSection));
+    }
+
+    private boolean isInvalidCountInMiddle(List<Section> foundSections) {
+        int count = foundSections.size();
+        return count < 1 || count > 2;
+    }
+
+    private boolean isSameWithDestinations(List<Section> foundSections) {
+        return values.size() > 2 && foundSections.size() == 2 &&
+            foundSections.containsAll(List.of(values.get(0), values.get(values.size() - 1)));
+    }
+
+    public Station getUpDestination() {
+        return values.get(0).getUpStation();
+    }
+
+    public Station getDownDestination() {
+        return values.get(values.size() - 1).getDownStation();
+    }
+
+    public List<Section> getValues() {
+        return List.copyOf(values);
+    }
+
+    public Section delete(Station station) {
+        validateDeletable();
+
+        if (station.equals(getUpDestination())) {
+            return values.remove(0);
+        }
+
+        if (station.equals(getDownDestination())) {
+            return values.remove(values.size() - 1);
+        }
+
+        return mergeTwoSections(findSectionToDelete(station));
+    }
+
+    private void validateDeletable() {
+        if (values.size() <= 1) {
+            throw new IllegalArgumentException("구간이 하나인 경우 삭제할 수 없습니다.");
+        }
+    }
+
+    private Section findSectionToDelete(Station station) {
+        return values.stream()
+            .filter(it -> it.contains(station))
+            .findFirst()
+            .orElseThrow(() -> new IllegalArgumentException("삭제할 역이 존재하지 않습니다."));
```

Comment:
메시지를 조금 더 명확히 할 수 있을 것 같아요. 역 자체가 존재하지 않는 것이 아니라 특정 노선의 구간(들)에 속하지 않은 상황이니까요.

Reply:
감사합니다!!! 상황에 알맞는 메시지를 반환하도록 수정해보았습니다.

```suggestion
            .orElseThrow(() -> new IllegalArgumentException("해당 노선 내에 삭제 요청된 역이 포함된 구간이 존재하지 않습니다."));
```